hooks/src/index.js

```diff
@@ -250,11 +250,24 @@ export function useContext(context) {
 	// This is probably not safe to convert to "!"
 	if (state._value == null) {
 		state._value = true;
-		provider.sub(currentComponent);
+		state._subscription = provider.sub(currentComponent, selector);
 	}
+
 	return provider.props.value;
 }
 
+/**
+ * @param {import('./internal').PreactContext} context
+ * @param {(c: any) => any} selector
+ */
+export function useContextSelector(context, selector) {
+	const contextValue = useContext(context, selector)
+	const state = getHookState(currentIndex, 11);
+	state._subscription[1] = selector;
```

Comment:
Since nothing currently assumes `useContext()` will ignore updates to `selector`, maybe we could just have it do this there?

```js
export function useContext(context, selector) {
	const provider = currentComponent.context[context._id];
	const state = getHookState(currentIndex++, selector ? 11 : 9);
	state._context = context;
	if (!provider) return context._defaultValue;
	const sub = state._subscription || (state._subscription = provider.sub(currentComponent));
	sub[1] = selector;
	return selector ? selector(provider.props.value) : provider.props.value;
}

export const useSelectedContext = useContext;
```

Reply:
Ohh I like that idea :+1:

Reply:
another option that might be worth checking (really just for size) would be fully duplicating the logic here into `useContext` / `useContextSelector` - duplication tends to be better when compressed, might be smaller than the two ternaries above.